mono/utils/mono-threads-coop.c

```diff
@@ -196,16 +198,16 @@ copy_stack_data (MonoThreadInfo *info, gpointer *stackdata_begin)
 }
 
 static gpointer
-mono_threads_enter_gc_safe_region_unbalanced_with_info (MonoThreadInfo *info, gpointer *stackdata);
+mono_threads_enter_gc_safe_region_unbalanced_with_info (MonoThreadInfo *info, MonoStackData *stackdata);
 
 gpointer
-mono_threads_enter_gc_safe_region (gpointer *stackdata)
+mono_threads_enter_gc_safe_region_internal (MonoStackData *stackdata)
```

Comment:
We still need `mono_threads_enter_gc_safe_region` (and `mono_threads_exit_gc_safe_region`) are still in the header, but don't have an implementation anymore?

Also the `thunks.exe` failure on coop is directly related - it tries to use these functions to move from GC Safe mode to GC Unsafe, and back.

Reply:
Agreed the failure is related and I will followup.
I will double check that I didn't remove anything, quite possible.